### PR TITLE
Full page invite token page during onboarding.

### PR DESCRIPTION
**Pull request by rahulgi**
Also forces all new members to create an identity video, regardless of whether they were invited via joint video.
Also fixes a bug in redirecting during onboarding.

screen

### CODE REVIEW

File: src/components/pages/LogIn.tsx [hunk 1]
```diff
@@ -446,7 +446,7 @@ class LogInView extends React.Component<LogInProps, LogInState> {
       const redirectParams = this.props.navigation.getParam("redirectParams");
       this.props.navigation.navigate(
         redirectTo,
-        ...(redirectParams ? redirectParams : {})
+        redirectParams ? redirectParams : {}
```
Comment (osdiab) [inline src/components/pages/LogIn.tsx]: meh bummer the types don't catch this, guess there's some leakage somewhere along the line?

File: src/components/pages/Onboarding/Onboarding.tsx [hunk 2]
```diff
@@ -171,6 +173,10 @@ class OnboardingView extends React.Component<OnboardingProps, OnboardingState> {
       this.steps.push(prevState.step);
     }
 
+    if (prevProps.isLoggedIn && !this.props.isLoggedIn) {
+      this._redirectIfLoggedOut();
```
Comment (tinaroh) [inline src/components/pages/Onboarding/Onboarding.tsx]: Should we return if this happens?

File: src/components/pages/Onboarding/Onboarding.tsx [hunk 3]
```diff
@@ -182,7 +188,18 @@ class OnboardingView extends React.Component<OnboardingProps, OnboardingState> {
       this.state.verifiedName &&
       this.state.emailAddress
     ) {
-      this._goToStep(this._validatedCameraStep());
+      this._goToStep(this._validatedInviteTokenStep());
```
Comment (tinaroh) [inline src/components/pages/Onboarding/Onboarding.tsx]: Should this case be removed now? This was for skipping the identity video if the user has a valid video already

**Comment by tinaroh** [comment]
It's a lot of text for a technically optional step...
Can we say: "Were you invited by someone?" in header text,
"No" "Yes"

and if they click yes show the text input?

File: src/components/pages/Onboarding/OnboardingCreateAccount.tsx [hunk 4]
```diff
@@ -67,17 +67,23 @@ class OnboardingCreateAccountView extends React.Component<
     const statusType = status ? status.status : undefined;
     switch (statusType) {
       case ApiCallStatusType.STARTED:
-        return <Text>Creating account...</Text>;
+        return <Text style={styles.paragraph}>Creating account...</Text>;
```
Comment (tinaroh) [inline src/components/pages/Onboarding/OnboardingCreateAccount.tsx]: Do we override the invite video when the user creates an identity video?

Comment (rahulgi) [reply src/components/pages/Onboarding/OnboardingCreateAccount.tsx]: nope

File: src/components/pages/Onboarding/InputInviteToken.tsx [hunk 5]
```diff
@@ -88,24 +90,37 @@ class InputInviteTokenView extends React.Component<
...
+                style={styles.input}
               />
               {this.state.errorMessage && (
                 <Text style={[{ color: colors.error }, sharedStyles.paragraph]}>
                   {this.state.errorMessage}
                 </Text>
               )}
               <Button
-                title={`Confirm`}
+                title={`Confirm invite token`}
```
Comment (markulrich) [inline src/components/pages/Onboarding/InputInviteToken.tsx]:
```suggestion
                title="Confirm invite token"
```

File: src/components/pages/Onboarding/Onboarding.tsx [hunk 6]
```diff
@@ -277,28 +294,30 @@ class OnboardingView extends React.Component<OnboardingProps, OnboardingState> {
...
-      ? // The new member does not need to take a verification video
-        // if they have a valid joint invite video.
-        OnboardingStep.CREATE_ACCOUNT
-      : OnboardingStep.CAMERA;
+  _validatedInviteTokenStep() {
+    return this.props.hasValidInviteToken && this.state.inviteVideoIsValid
+      ? // The new member is not requested to input an invite token
+        // if they already have a valid one.
+        // Note, Previously we would send the user directly to CREATE_ACCOUNT
+        // if the onboarding video was joint. Now we require an identity video from
```
Comment (markulrich) [inline src/components/pages/Onboarding/Onboarding.tsx]: I do like the idea of everyone having an identity video, but I also don't want to add it as a mandatory extra step for someone who already made a video with their friend as that feels very heavy for this early stage ... would be nice to let people see what Raha is about ASAP and then allow and encourage members to update their identity video at any time.

File: src/components/pages/LogIn.tsx (hunk 1 above)
Comment (rahulgi) [reply src/components/pages/LogIn.tsx]: I think the leakage is in React-Navigation `getParam` returning `any` type. XD

File: src/components/pages/Onboarding/Onboarding.tsx (hunk 2 above)
Comment (rahulgi) [reply src/components/pages/Onboarding/Onboarding.tsx]: sounds like a good idea!

File: src/components/pages/Onboarding/Onboarding.tsx (hunk 6 above)
Comment (rahulgi) [reply src/components/pages/Onboarding/Onboarding.tsx]: let's give it a shot and can easily roll back if it turns out to be a sticking point

File: src/components/pages/Onboarding/InputInviteToken.tsx (hunk 5 above)
Comment (rahulgi) [reply src/components/pages/Onboarding/InputInviteToken.tsx]: 👍

File: src/components/pages/Onboarding/Onboarding.tsx (hunk 3 above)
Comment (rahulgi) [reply src/components/pages/Onboarding/Onboarding.tsx]: This is still used to check whether the video for the invite token that the user input exists before advancing.

**Pull request merged**